music/src/core/player.ts

```diff
@@ -328,9 +328,89 @@ export class SoundFontPlayer extends BasePlayer {
 }
 
 /**
- * A `NoteSequence` player based on Tone.js that includes a click track.
+ * A base class for providing arbitrary callbacks for each note played.
  */
-export class PlayerWithClick extends Player {
+export class BasePlayerCallback {
```

Comment:
make this an abstract class with abstract methods signatures

Reply:
done